Iced/Intel/Encoder.cs

```diff
@@ -632,6 +632,63 @@ public sealed class Encoder {
 			EncoderFlags |= (EncoderFlags)((regNum & 0x10) << (9 - 4));
 		}
 
+		internal void AddReg(ref Instruction instr, int operand, Register regLo, Register regHi) {
+			var opKind = instr.GetOpKind(operand);
+			if (opKind != OpKind.Register) {
+				ErrorMessage = $"Operand {operand}: Expected a register, but opKind is {opKind}";
+				return;
+			}
+
+			var reg = instr.GetOpRegister(operand);
+			uint regNum = (uint)(reg - regLo);
```

Comment:
Can the logic below here be simplified. Have a look at AddModRMRegister, it can handle all GPR8, GPR16, GPR32, GPR64 and contains only 2 if statements.